packages/core/lib/commands/compile.js

```diff
@@ -98,28 +97,55 @@ const command = {
   },
 
   listVersions: async function (options) {
-    const {CompilerSupplier} = require("@truffle/compile-solidity");
+    const { CompilerSupplier } = require("@truffle/compile-solidity");
+    const { asyncTake } = require("iter-tools");
+
     const supplier = new CompilerSupplier({
-      solcConfig: options.compilers.solc,
+      solcConfig: {
+        ...options.compilers.solc,
+        docker: options.list === "docker"
+      },
       events: options.events
     });
 
     const log = options.logger.log;
     options.list = options.list.length ? options.list : "releases";
 
+    const {
+      latestRelease,
+      releases,
+      prereleases
+    } = await supplier.list();
```

Comment:
In this file, that condition won't happen - it uses either VersionRange or Docker only. Inside the implementation of `supplier.list()`, there's a check.

It's a bit weird, but I've addressed this in #4224 by making it explicit whether a strategy supplies listing functionality; or see #4225 for implementation.

Reply:
It looks to me like that condition can happen?  If the solc version is set to `"native"` or a local soljson (and `docker` is not set to true), it looks like the CompilerSupplier will be created with the corresponding loading strategy.  What am I missing?

I do see that the check you mentioned exists, so that we at least throw a proper error, but this should be a can't-happen and right now it looks like it isn't?

Reply:
Added [a note](https://github.com/trufflesuite/truffle/pull/4225/files#r688157235) to ensure this gets addressed in #4225.

Reply:
I think that it would be better to address this here since the problem is being introduced here?

Reply:
I agree with Harry. If the problem is introduced in this PR it should be fixed here.

Reply:
Sound reasoning! One sec :)

Reply:
OK pushed. It's not as clean as my solution offered in #4225, because in this PR, Docker still requires a version. (In the other PR, I fixed it to conform.)

Let me know if this is alright!

Reply:
The hacky solution is fine by me.